### PR TITLE
Feature/d3 asim 2115

### PR DESCRIPTION
**Pull request by hannesdiedrich**

### CODE REVIEW

File: d3a_api_client/redis_device.py [hunk 1]
```diff
@@ -11,11 +11,15 @@ def __init__(self, device_id, autoregister=True, redis_url='redis://localhost:63
 
     def _on_register(self, msg):
         message = json.loads(msg["data"])
+        if not self._is_trans_id_matching(message):
+            return
```
Comment (spyrostz) [inline d3a_api_client/redis_device.py]: I think it is better if you raise an exception here. That way the user will be notified that the registration failed, and the client will exit.

File: d3a_api_client/redis_device.py [hunk 2]
```diff
@@ -11,11 +11,15 @@ def __init__(self, device_id, autoregister=True, redis_url='redis://localhost:63
...
         logging.info(f"Client was registered to the device: {message}")
         self.is_active = True
 
     def _on_unregister(self, msg):
         message = json.loads(msg["data"])
+        if not self._is_trans_id_matching(message):
```
Comment (spyrostz) [inline d3a_api_client/redis_device.py]: Same here

File: d3a_api_client/redis_device.py (hunk 1 above)
Comment (hannesdiedrich) [reply d3a_api_client/redis_device.py]: You are right! There is only ONE register request in this channel. Will add it.

Comment (hannesdiedrich) [reply d3a_api_client/redis_device.py]: But this applies also to the rest client, right? Will add it there too.

Comment (spyrostz) [reply d3a_api_client/redis_device.py]: Applies to the rest client indeed. Thanks

File: d3a_api_client/redis.py [hunk 3]
```diff
@@ -264,6 +261,13 @@ def executor_function():
             self.on_finish(message)
         self.executor.submit(executor_function)
 
+    def _check_buffer_message_matching_command_and_id(self, message):
+        transaction_id = message["transaction_id"]
+        if not any(command == "register" and "transaction_id" in data and
+                   data["transaction_id"] == transaction_id
+                   for command, data in self._blocking_command_responses.items()):
+            raise Exception("There is no matching command response in _blocking_command_responses.")
```
Comment (faizan2590) [inline d3a_api_client/redis.py]:
```suggestion
            raise RedisAPIException("There is no matching command response in _blocking_command_responses.")
```

File: d3a_api_client/redis.py [hunk 4]
```diff
@@ -9,7 +10,7 @@
 from d3a_api_client.constants import MAX_WORKER_THREADS
 
 root_logger = logging.getLogger()
-root_logger.setLevel(logging.ERROR)
+root_logger.setLevel(logging.INFO)
```
Comment (faizan2590) [inline d3a_api_client/redis.py]: Any particular reason to change the log level?

File: d3a_api_client/redis.py [hunk 5]
```diff
@@ -264,6 +261,13 @@ def executor_function():
             self.on_finish(message)
         self.executor.submit(executor_function)
 
+    def _check_buffer_message_matching_command_and_id(self, message):
+        transaction_id = message["transaction_id"]
```
Comment (faizan2590) [inline d3a_api_client/redis.py]: Can you also check and raise an appropriate exception if `transaction_id` is None or if `transaction_id` key not part of message ?

Comment (hannesdiedrich) [reply d3a_api_client/redis.py]: yes! Good idea! Thanks!

File: d3a_api_client/redis.py (hunk 4 above)
Comment (hannesdiedrich) [reply d3a_api_client/redis.py]: No, this is a remnant of testing. Will change it back.

**Pull request merged**